### PR TITLE
feat: block quote

### LINKED ISSUE #414: New component: Blockquote
Purpose: Create new Blockquote component

InVision: https://cengage.invisionapp.com/share/GM123CETDTN4

image

### AC
- Content
-- Able to provide the quote (required)
-- Able to provide source attribution (optional)
- Style
-- Default styles
--- Quote = body/large, productive font style
--- Source attribution = body/medium, productive font style, subdued color
--- Bar on the left
---- 4px wide, background-color = #DFDFDF
--- Padding = 12px top/bottom, 16px left/right
-- Able to customize the type style of both quote and source attribution, including changing to Expressive or Narrative
-- Able to customize the color of the bar
- Inverse
-- Provide default inverse version as well.

image

### PR DESCRIPTION
**Pull request by chris-cedrone-cengage**

### CODE REVIEW

**Review by orion-cengage — CHANGES_REQUESTED** [review]
1. Can we add a space in between the m-dash and the attribution?
image

2.  Since the quotation marks are built into the component, can we make sure we're using smart quotes, so they look less like this:
image

and more like this? 
image

I think it's a matter of using these:
image

It will be hard to tell with Open Sans, but should be easier to see on Noto Serif.

3. Blockquote is one word, so please fix instances like this:
image

and 
image

**Review by aaronmaturen — CHANGES_REQUESTED** [review]
https://css-tricks.com/quoting-in-html-quotations-citations-and-blockquotes/#hey-what-about-the-figure-element

we should probably use `figure` `blockquote` and `figcaption` for these...

**Review by orion-cengage — APPROVED** [review]

File: packages/react-magma-dom/src/components/BlockQuote/BlockQuote.test.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import { axe } from 'jest-axe';
+import { BlockQuote, BlockQuoteItem } from '.';
+import { magma } from '../../theme/magma';
+import { render } from '@testing-library/react';
+
+const TEXT = 'Test Text';
+
+describe('BlockQuote', () => {
+  it('should render the visually hidden component', () => {
```
Comment (rcgillet) [inline packages/react-magma-dom/src/components/BlockQuote/BlockQuote.test.js:10]: test description doesn't make sense.

File: packages/react-magma-dom/src/components/BlockQuote/BlockQuote.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      ...rest
+    } = props;
+    const theme = React.useContext(ThemeContext);
+    const isInverse = useIsInverse(isInverseProp);
+
+    return (
+      <InverseContext.Provider value={{ isInverse }}>
+        <StyledBlockQuote
+          borderStyle={
+            magma.colors[borderStyle] ? magma.colors[borderStyle] : borderStyle
```
Comment (rcgillet) [inline packages/react-magma-dom/src/components/BlockQuote/BlockQuote.tsx:54]:
```suggestion
            magma.colors[borderStyle] || borderStyle
```

File: packages/react-magma-dom/src/components/BlockQuote/BlockQuoteItem.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  const theme = React.useContext(ThemeContext);
+  const isInverse = useIsInverse();
+
+  return (
+    <StyledBlockQuoteItem
+      as={hasAttribution ? 'figcaption' : 'blockquote'}
+      color={
+        hasAttribution
+          ? color || TypographyColor.subdued
+          : color || TypographyColor.default
```
Comment (rcgillet) [inline packages/react-magma-dom/src/components/BlockQuote/BlockQuoteItem.tsx:81]:
```suggestion
        color || hasAttribution ? TypographyColor.subdued : TypographyColor.default
```

File: packages/react-magma-dom/src/components/BlockQuote/BlockQuoteItem.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      }
+      theme={theme}
+      hasAttribution={hasAttribution}
+      isInverse={isInverse}
+      ref={ref}
+      data-testid={props.testId}
+      visualStyle={
+        hasAttribution
+          ? visualStyle || TypographyVisualStyle.bodyMedium
+          : visualStyle || TypographyVisualStyle.bodyLarge
```
Comment (rcgillet) [inline packages/react-magma-dom/src/components/BlockQuote/BlockQuoteItem.tsx:91]:
```suggestion
      visualStyle={
        visualStyle || hasAttribution ? TypographyVisualStyle.bodyMedium : TypographyVisualStyle.bodyLarge
```

**Review by rcgillet — CHANGES_REQUESTED** [review]

**Review by rcgillet — APPROVED** [review]

**Pull request merged**